lib/netext/http.go

```diff
@@ -0,0 +1,143 @@
+package netext
+
+import (
+	"bytes"
+	"encoding/base64"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"sync"
+
+	"github.com/ThomsonReutersEikon/go-ntlm/ntlm"
+	"github.com/pkg/errors"
+)
+
+type HTTPTransport struct {
+	*http.Transport
+
+	mu        sync.Mutex
+	authCache map[string]bool
+}
+
+func NewHTTPTransport(transport *http.Transport) *HTTPTransport {
+	return &HTTPTransport{
+		Transport: transport,
+		authCache: make(map[string]bool),
+	}
+}
+
+func (t *HTTPTransport) RoundTrip(req *http.Request) (res *http.Response, err error) {
```

Comment:
I agree with you, but I couldn't find a different way. At least if it is not an NTLM request, everything else works as it is supposed to be.

Reply:
Won't we plug the digest auth here as well at some point? So it doesn't make two requests every time?

Reply:
Given that digest doesn't require authentication by connection, I guess we could instead of putting this logic inside the round tripper, we should do it where it is supposed to be, inside the client.

Reply:
It might be confusing having two places with different authentications, but at least we are minimizing what we doing wrong.

Reply:
It doesn't require authentication by connection, but if we want to have an optimized digest authentication, we'll probably have to cache the authorization header between different requests with the same domain and credentials. I have no idea what the best way to do that is...

Reply:
To accomplish that, we do the same we did in ntlm but at the client level, it would require a custom Client. It can be as simple as the ntlm, check if it is digest, otherwise, do nothing and call the default implementation.